### PR TITLE
MULE-16943: SDK operations optimizations: Avoid non-blocking overhead for known blocking calls

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java [hunk 1]
```diff
@@ -483,14 +507,18 @@ protected boolean usesDynamicConfiguration() {
    * returns an empty value.
    */
   protected Optional<ConfigurationInstance> getStaticConfiguration() {
-    if (!requiresConfig.get() || (isConfigurationSpecified() && configurationProvider.get().isDynamic())) {
+    if (!requiresConfig()) {
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java]: isn't this changing the behavior?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/DefaultExecutionMediator.java [hunk 2]
```diff
@@ -167,15 +167,15 @@ public void error(Throwable e) {
...
+    executeCommand.accept(callbackDecorator);
   }
 
   private void executeWithRetry(ExecutionContextAdapter<M> context,
                                 RetryPolicyTemplate retryPolicy,
                                 List<Interceptor> executedInterceptors,
                                 Consumer<ExecutorCallback> executeCommand,
-                                BiConsumer<ExecutorCallback, Object> onSuccess,
-                                BiConsumer<ExecutorCallback, Throwable> onError,
+                                //BiConsumer<ExecutorCallback, Object> onSuccess,
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/DefaultExecutionMediator.java]: remove

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java [hunk 3]
```diff
@@ -72,34 +70,33 @@ public OAuthOperationMessageProcessor(ExtensionModel extensionModel,
...
 
         OAuthConnectionProviderWrapper connectionProvider = getOAuthConnectionProvider(operationContext);
         if (connectionProvider == null) {
-          delegate.error(e);
+          callback.error(e);
           return;
         }
 
-        Reference<Optional<String>> resourceOwnerIdReference = new Reference<>(empty());
+        Reference<Optional<String>> resourceOwnerIdReference = new Reference<>(Optional.empty());
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java]: static import

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java (hunk 1 above)
Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java]: nope, not at all. On the contrary. this is not the first version of this optimization. This is the one that fixed all the tests that I broke :p

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java (hunk 3 above)
Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java]: can't. it conflicts with other imports

**Pull request merged**